public/app/features/alerting/unified/AlertsFolderView.tsx

```diff
@@ -0,0 +1,216 @@
+import { css } from '@emotion/css';
+import { GrafanaTheme2, SelectableValue } from '@grafana/data';
+import { Stack } from '@grafana/experimental';
+import { Card, FilterInput, Icon, Pagination, Select, TagList, useStyles2 } from '@grafana/ui';
+import { DEFAULT_PER_PAGE_PAGINATION } from 'app/core/constants';
+import { FolderState } from 'app/types';
+import { CombinedRule } from 'app/types/unified-alerting';
+import { isEqual, orderBy, uniqWith } from 'lodash';
+import React, { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { useDebounce } from 'react-use';
+import { useCombinedRuleNamespaces } from './hooks/useCombinedRuleNamespaces';
+import { usePagination } from './hooks/usePagination';
+import { useURLSearchParams } from './hooks/useURLSearchParams';
+import { fetchPromRulesAction, fetchRulerRulesAction } from './state/actions';
+import { labelsMatchMatchers, matchersToString, parseMatcher, parseMatchers } from './utils/alertmanager';
+import { GRAFANA_RULES_SOURCE_NAME } from './utils/datasource';
+import { createViewLink } from './utils/misc';
+
+interface Props {
+  folder: FolderState;
+}
+
+enum SortOrder {
+  Ascending = 'alpha-asc',
+  Descending = 'alpha-desc',
+}
+
+type SortOrderType = `${SortOrder}`;
+
+const sortOptions: Array<SelectableValue<SortOrderType>> = [
+  { label: 'Alphabetically [A-Z]', value: SortOrder.Ascending },
+  { label: 'Alphabetically [Z-A]', value: SortOrder.Descending },
+];
+
+export const AlertsFolderView = ({ folder }: Props) => {
+  const styles = useStyles2(getStyles);
+  const dispatch = useDispatch();
+
+  const onTagClick = (tagName: string) => {
+    const matchers = parseMatchers(labelFilter);
+    const tagMatcherField = parseMatcher(tagName);
+    const uniqueMatchers = uniqWith([...matchers, tagMatcherField], isEqual);
+    const matchersString = matchersToString(uniqueMatchers);
+    setLabelFilter(matchersString);
+  };
+
+  useEffect(() => {
+    dispatch(fetchPromRulesAction({ rulesSourceName: GRAFANA_RULES_SOURCE_NAME }));
+    dispatch(fetchRulerRulesAction({ rulesSourceName: GRAFANA_RULES_SOURCE_NAME }));
+  }, [dispatch]);
+
+  const combinedNamespaces = useCombinedRuleNamespaces(GRAFANA_RULES_SOURCE_NAME);
+  const { nameFilter, labelFilter, sortOrder, setNameFilter, setLabelFilter, setSortOrder } =
+    useAlertsFolderViewParams();
+
+  const matchingNamespace = combinedNamespaces.find((namespace) => namespace.name === folder.title);
+  const alertRules = matchingNamespace?.groups[0]?.rules ?? [];
+
+  const filteredRules = filterAndSortRules(alertRules, nameFilter, labelFilter, sortOrder ?? 'alpha-asc');
+
+  const showNoResultsText = alertRules.length === 0 || filteredRules.length === 0;
```

Comment:
nit: `hasNoResults` would make a bit more sense here; the current variable one already includes what the check is going to be used for :)